designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md

```diff
@@ -0,0 +1,414 @@
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+  </ItemGroup>
+
+</Project>
+```
+
+### Remove a package from a project
+
+In this scenario, there is a `Directory.Packages.props` file which defines a version for `PackageA` and `PackageB`, and a project `Project1` has a `PackageReference` to both `PackageA` and `PackageB`. The customer uninstalls `PackageA`, and the `PackageReference` must be removed from `Project1`, leaving `PackageB`. `Directory.Packages.props` should remain unmodified, as other projects might be referencing `PackageA`.
+
+#### Before
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+    <PackageVersion Include="PackageB" Version="4.5.6" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageA" />
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+#### Gesture
+
+There are at least 3 gestures which may meet this scenario
+
+1. Install `PackageA` using Package Manager UI. The customer will not be able to select the version.
+2. Install `PackageA` using Package Manager Console.
+3. NuGet's `InstallPackageAsync` API is called, to install `PackageA`. For example, triggered from a Roslyn analyzer code fix.
+
+#### After
+
+`$(repo_root)\Directory.Packages.props`
+
+```xml
+<Project>
+  <ItemGroup>
+    <PackageVersion Include="PackageA" Version="1.2.3" />
+    <PackageVersion Include="PackageB" Version="4.5.6" />
+  </ItemGroup>
+</Project>
+```
+
+`$(repo_root)\Project1\Project1.csproj`
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="PackageB" />
+  </ItemGroup>
+
+</Project>
+```
+
+### Update the version
+
+In this scenario, there is a `Directory.Packages.props` file which defines a version for `PackageA` and `PackageB`. There are two projects, `Project1` has a `PackageReference` to `PackageA`, and `Project2` has a `PackageReference` to `PackageB`. The customer updates `PackageA` to a different version. The `Directory.Packages.props` must be modified to contain the new version. All projects (two in this scenario) must be restored. `Project1` must be restored as it directly references `PackageA`. `Project2` must be restored, because `PackageB` might have a dependency on `PackageA`, and therefore the `PackageA`'s version needs to be pinned.
```

Comment:
If the ProjectA is defined in a imported Directory.Packages.props we need to decide this action what result will have. 

Is the parent that will have the version updated. 
Or a different simpler option will be to act as an addition and add the new version to the the closest Directory.Packages.props.